components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/auth/service/AuthServiceResponseWrapper.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.model.auth.service;
+
+import org.wso2.carbon.identity.application.authentication.framework.exception.auth.service.AuthServiceException;
+import org.wso2.carbon.identity.application.authentication.framework.model.CommonAuthResponseWrapper;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import org.wso2.carbon.identity.application.authentication.framework.util.auth.service.AuthServiceConstants;
+import org.wso2.carbon.identity.application.authentication.framework.util.auth.service.AuthServiceUtils;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * A HttpServletResponseWrapper wrapper to be used in authentication service.
+ */
+public class AuthServiceResponseWrapper extends CommonAuthResponseWrapper {
+
+    public AuthServiceResponseWrapper(HttpServletResponse response) {
+
+        super(response);
+    }
+
+    /**
+     * Get the value of the authenticators query param in the redirect URL.
+     *
+     * @return String of authenticators and related IdPs engaged in the authentication flow.
+     * @throws AuthServiceException
+     */
+    public String getAuthenticators() throws AuthServiceException {
+
+        Map<String, String> queryParams = AuthServiceUtils.extractQueryParams(getRedirectURL());
+        return queryParams.get(AuthServiceConstants.AUTHENTICATORS);
+    }
+
+    /**
+     * Get the flowId related to the authentication flow.
+     * This is the sessionDataKey of the authentication flow.
+     *
+     * @return String of flowId.
+     * @throws AuthServiceException
+     */
+    public String getFlowId() throws AuthServiceException {
```

Comment:
any reason to use this as `flowId` instead of the existing session data key term? We should try to keep the same to avoid any confusions (at least in internal classes, in API level params it would be fine to use a matching name for the api)

Reply:
will change this. This was done thinking of what we want to send outside.